docker.py

```diff
@@ -5,75 +5,114 @@
 from subprocess import Popen, PIPE
 from socket import getfqdn
 
-singularity = True
-if any('grid-workernode-c7' in arg for arg in sys.argv):
-    singularity = True
 
-gateway = False
-if any('atl' in arg for arg in sys.argv) or any('cms' in arg for arg in sys.argv) or any('lhcb' in arg for arg in sys.argv):
-    gateway = True
+def gateway():
+    gateway_needed = False
+    if any('atl' in arg for arg in sys.argv) or \
+       any('cms' in arg for arg in sys.argv) or \
+       any('lhcb' in arg for arg in sys.argv):
+        gateway_needed = True
+    return gateway_needed and not os.path.isfile('/etc/nogateway')
 
-count = 0
-dargs = ['/usr/bin/sudo', '/usr/bin/docker']
+def args_create(argv):
+    """
+    Build the new list of command line arguments for command
+        docker create
+    """
+    dargs = []
 
-if os.environ.get('DOCKER_WRAPPER_DEBUG'):
-    dargs = ['/bin/echo'] + dargs
+    # Allow singularity to work inside of Docker containers
+    dargs.append('-eSINGULARITY_BINDPATH=/etc/hosts')
+    dargs.append('--cap-add=SYS_ADMIN')
+    dargs.append('--cap-add=DAC_OVERRIDE')
+    dargs.append('--cap-add=SETUID')
+    dargs.append('--cap-add=SETGID')
+    dargs.append('--cap-add=SYS_CHROOT')
+    # ATLAS fix for 21.0.XX release errors with frontier
+    dargs.append('--env=SINGULARITYENV_FRONTIER_LOG_FILE=frontier.log')
+
```

Comment:
We may want to be aware of a potential regression with https://github.com/stfc/ral-htcondor-tools/pull/26 Will depend on outcome of ATLAS failed jobs